### PR TITLE
ci: golangci

### PR DESCRIPTION
**Pull request by faddat**
Closes: #XXX

## Context and purpose of the change

This PR improves linting and fixes the lints that made the ignores needed, either by specifically ignoring certain lines of code for certain 
linters, or by fixing the linting issues. 

## Brief Changelog

* always use latest golangci-lint
* golangci-lint settings
* minor linting improvements

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.18
+          go-version: 1.19
```
Comment (asalzmann) [inline .github/workflows/lint.yml:18]: we're still using go 1.18, does changing the version here just apply to the version of go that executes the linter?

File: .golangci.yml [hunk 2]
```diff
@@ -1,9 +1,104 @@
...
-    - text: "SA1019: package github.com/golang/protobuf/proto is deprecated: Use the \"google.golang.org/protobuf/proto\" package instead."
-      linters:
-        - staticcheck
-    - text: "SA1019: clientCtx.JSONCodec is deprecated: Codec codec will be changed to Codec: codec.Codec"
-      linters:
-        - staticcheck
+run:
+  timeout: 5m
+  disable-all: true
+  enable:
```
Comment (asalzmann) [inline .golangci.yml:4]: How did you decide on this list of lint checks to run?

File: cmd/strided/genaccounts.go [hunk 3]
```diff
@@ -43,8 +42,8 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
 		Args: cobra.ExactArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			clientCtx := client.GetClientContextFromCmd(cmd)
-			depCdc := clientCtx.JSONCodec
```
Comment (asalzmann) [inline cmd/strided/genaccounts.go:46]: What's the purpose of this change?

**Review by asalzmann — COMMENTED** [review]
looks good overall! just a few minor comments. thanks for cleaning this up!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (faddat) [reply .github/workflows/lint.yml:18]: That's correct

File: .golangci.yml (hunk 2 above)
Comment (faddat) [reply .golangci.yml:4]: Well I have written that list on many many chains in Cosmos and it seems to be approximately the correct ones

Comment (faddat) [reply .golangci.yml:4]: Oh hey I'm sorry this is probably too long of a list

Comment (asalzmann) [reply .golangci.yml:4]: Gotcha, this makes sense

**Comment by asalzmann** [comment]
@faddat is this good to merge, or should we thin out the list?

**Comment by faddat** [comment]
Yeah it's good to merge but I need to update the branch so I'll get on that in a moment

**Review by asalzmann — APPROVED** [review]
